pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -686,8 +692,12 @@ async def _handle_model_response(
                 else:
                     messages.append(self._unknown_tool(call.tool_name))
 
-            with _logfire.span('running {tools=}', tools=[t.get_name() for t in tasks]):
-                messages += await asyncio.gather(*tasks)
+            with _logfire.span('running {tools=}', tools=[t.get_name() for t in tasks]) as span:
+                try:
+                    messages += await asyncio.gather(*tasks)
+                except EndAgentRun as e:
+                    span.set_attribute('end_agent_run', e.tool_name)
+                    return _MarkFinalResult(data=e.result)
```

Comment:
I guess we need to update `run_stream` too.

Reply:
I think this is sorted now